core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/UniqueIDReadOnlyLDAPUserStoreManager.java

```diff
@@ -2195,4 +2248,570 @@ protected void processAttributesAfterRetrievalWithID(String userID, Map<String,
         }
     }
 
+    /**
+     * Do LDAP paginated search and return user objects as a list.
+     *
+     * @param ldapContext             LDAP connection context.
+     * @param ldapSearchSpecification Contains LDAP context search parameters.
+     * @param pageSize                Number of results per page.
+     * @param offset                  Start index.
+     * @param expressionConditions    List of input expressions.
+     * @return List of users.
+     * @throws UserStoreException
+     */
+    private List<User> performLDAPSearch(LdapContext ldapContext, LDAPSearchSpecification ldapSearchSpecification,
```

Comment:
Can't we make the super as protected and reuse this?

Reply:
These newly added methods differ from super class as super class methods process a list of usernames (Strings) and new methods process a list of User objects.